### PR TITLE
Extended readme.md

### PR DESCRIPTION
**Pull request by orangeDK**
Added a new chapter about flashing an image for Charge Control C and Charge Control M products.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ For problems and inquiries: https://tickets.in-tech-smartcharging.com/servicedes
 3.2 [Setting up the Yocto build environment](#Setting)  
 3.3 [Adding or removing layers](#addorremove)  
 3.4 [Building an Image](#build)  
+3.5 [Flashing an image](#flash)
```
Comment (barsnick) [inline README.md]: "Image" (capital) in 3.4, vs. "image" (non-capital) in 3.5.

File: README.md [hunk 2]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
 2. Execute `source yocto/source/oe-init-build-env build` which initializes the build environment and changes the directory to `yocto/build`.
 3. Execute `bitbake core-image-minimal` to build the image.
 
-The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>`.
+The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>` have a filename similar to `core-image-minimal-platform-1234567890-rootfs.ext4.gz`
+
+### Flashing an image <a name="flash"></a>
+
+The internal storage of the products Charge Control C and M is divided into several partitions and uses rauc to handle updates. If you are interested in adding rauc to your own image you can take a look at raucs documentation https://rauc.readthedocs.io/en/latest/integration.html#yocto .To flash an image it is important to identify the currently not in use partition because this will be our target partition. The following tables describe the partitioning for each Charge Control M and Charge Control C:
```
Comment (mooraby) [inline README.md]: Would prefer to have RAUC in capital letters.
Also "RAUC's documentation"

File: README.md [hunk 3]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
...
+rauc status
+```
+2. Copy your image to `/srv` using any client capable of SFTP.
+3. In the `/srv` directory execute the following command to flash your image to `/dev/mmcblk0p2`
+```bash
+gzcat core-image-minimal-tarragon-1234567890-rootfs.ext4.gz > /dev/mmcblk0p2
+```
+4. Mount `/dev/mmcblk0p2` and replace `/sbin/init` binary file with the following commands. (Omitting this step will format logging and data partitions)
+```bash
+mount /dev/mmcblk0p2 /mnt
```
Comment (mooraby) [inline README.md:217]: not sure if we need sudo for mount and umount on the target platform

File: README.md [hunk 4]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
...
+| `/dev/mmcblk0p1` | `1 GB` | Root file system A |
+| `/dev/mmcblk0p2` | `1 GB` | Root file system B |
+| `/dev/mmcblk0p3` |  | Extended Partition Container |
+| `/dev/mmcblk0p5` | `1 GB` | Data Partition(/srv). This partition can be accessed by both file systems. |
+| `/dev/mmcblk0p6` | `128 MB` | Logging file system A (/var/log) |
+| `/dev/mmcblk0p7` | `128 MB` | Logging file system B (/var/log) |
+
+1. To Identify the currently booted partition execute the following command and look for the partition which is marked as `good`. For this example we will be on partition `/dev/mmcblk0p1` on a Charge Control C.
+```bash
+rauc status
```
Comment (mooraby) [inline README.md:187]: It would be better if an example of `rauc status` output is added

File: README.md [hunk 5]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
 2. Execute `source yocto/source/oe-init-build-env build` which initializes the build environment and changes the directory to `yocto/build`.
 3. Execute `bitbake core-image-minimal` to build the image.
 
-The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>`.
+The resulting image will be found in `yocto/build/tmp/deploy/image/<machine>` have a filename similar to `core-image-minimal-platform-1234567890-rootfs.ext4.gz`
```
Comment (mhei) [inline README.md]: I suggest: ... will be found in ..., (comma here before) having a filename...

File: README.md [hunk 6]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
...
+
+### Flashing an image <a name="flash"></a>
```
Comment (mhei) [inline README.md]: here too: uppercase I in image

File: README.md (hunk 2 above)
Comment (mhei) [inline README.md]: please use always uppercase "RAUC" since it is an abbreviation

File: README.md [hunk 7]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
...
+
+The internal storage of the products Charge Control C and M is divided into several partitions and uses rauc to handle updates. If you are interested in adding rauc to your own image you can take a look at raucs documentation https://rauc.readthedocs.io/en/latest/integration.html#yocto .To flash an image it is important to identify the currently not in use partition because this will be our target partition. The following tables describe the partitioning for each Charge Control M and Charge Control C:
+
+**Charge Control M:**
+| **`Partition`** | **`Size`**  | Description |
```
Comment (mhei) [inline README.md]: formatting nitpicks: partition and size is both, marked as bold and code; description neither nor, I'd suggest to drop both

File: README.md [hunk 8]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
...
+| `/dev/mmcblk0p1` | `3 MB` | Bootloader Partition |
+| `/dev/mmcblk0p2` | `1 GB` | Root file system A |
+| `/dev/mmcblk0p3` | `1 GB` | Root file system B |
+| `/dev/mmcblk0p4` |  | Extended Partition Container |
+| `/dev/mmcblk0p5` | `1 GB` | Data Partition(/srv). This partition can be accessed by both file systems. |
+| `/dev/mmcblk0p6` | `128 MB` | Logging file system A (/var/log) |
+| `/dev/mmcblk0p7` | `128 MB` | Logging file system B (/var/log) |
+
+**Charge Control C:**
+| **`Partition`** | **`Size`**  | Description |
```
Comment (mhei) [inline README.md]: same as above

File: README.md [hunk 9]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
...
+|--|--|--|
+| `/dev/mmcblk0p1` | `1 GB` | Root file system A |
+| `/dev/mmcblk0p2` | `1 GB` | Root file system B |
+| `/dev/mmcblk0p3` |  | Extended Partition Container |
+| `/dev/mmcblk0p5` | `1 GB` | Data Partition(/srv). This partition can be accessed by both file systems. |
+| `/dev/mmcblk0p6` | `128 MB` | Logging file system A (/var/log) |
+| `/dev/mmcblk0p7` | `128 MB` | Logging file system B (/var/log) |
+
+1. To Identify the currently booted partition execute the following command and look for the partition which is marked as `good`. For this example we will be on partition `/dev/mmcblk0p1` on a Charge Control C.
```
Comment (mhei) [inline README.md]: small i in identify

File: README.md [hunk 10]
```diff
@@ -154,7 +155,54 @@ export BB_ENV_PASSTHROUGH_ADDITIONS="PROJECT MACHINE"
...
+2. Copy your image to `/srv` using any client capable of SFTP.
+3. In the `/srv` directory execute the following command to flash your image to `/dev/mmcblk0p2`
+```bash
+gzcat core-image-minimal-tarragon-1234567890-rootfs.ext4.gz > /dev/mmcblk0p2
+```
+4. Mount `/dev/mmcblk0p2` and replace `/sbin/init` binary file with the following commands. (Omitting this step will format logging and data partitions)
+```bash
+mount /dev/mmcblk0p2 /mnt
+mv /mnt/sbin/init.orig /mnt/sbin/init
+umount /dev/mmcblk0p2 -l
```
Comment (mhei) [inline README.md]: I'd not recommend to use option -l here, the umount should work without - if not, then something is wrong

**Review by mhei — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (orangeDK) [reply README.md:217]: Not necessary.

**Review by mooraby — APPROVED** [review]

**Pull request merged**